indexing-service/src/main/java/org/apache/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -227,6 +227,34 @@ public Response isLeader()
     }
   }
 
+  @GET
+  @Path("/lockedIntervals")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getTaskLockedIntervals(@Context HttpServletRequest request)
+  {
+    // Perform authorization check
+    final ResourceAction resourceAction = new ResourceAction(
+        new Resource("lockedIntervals", ResourceType.STATE),
+        Action.READ
+    );
+    final Access authResult = AuthorizationUtils
+        .authorizeResourceAction(request, resourceAction, authorizerMapper);
+    if (!authResult.isAllowed()) {
+      throw new WebApplicationException(
+          Response.status(Response.Status.FORBIDDEN)
+                  .entity(StringUtils.format("Access-Check-Result: %s", authResult.toString()))
+                  .build()
+      );
+    }
+
+    // Build the response
+    final LockedIntervalsResponse response = new LockedIntervalsResponse(
```

Comment:
Can the API just return Map<String, DatasourceIntervals> (removing the need for another class)?

Reply:
Response.ok(taskStorageQueryAdapter.getLockedIntervals()).build()

Reply:
Fixed.